src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -102,8 +102,11 @@
     /**
      * Sets the list of pinned endpoints for this endpoint.
      * @param newPinnedEndpoints the set of pinned endpoints.
+     * @return true if the underlying set of pinned endpoints has changed, false
+     * otherwise. The return value was introduced to enable overrides to
+     * act upon the underlying set changing.
      */
-    public void pinnedEndpointsChanged(Set<String> newPinnedEndpoints)
+    public boolean pinnedEndpointsChanged(Set<String> newPinnedEndpoints)
```

Comment:
Isn't the purpose of the `PINNED_ENDPOINTS_PROPERTY_NAME` event already to notify when the list of pinned endpoints has changed?

Reply:
@brianh5 that thought crossed my mind and I can definitely leverage this functionality if you think it's cleaner, i.e. have the Endpoint instance register for events generated by its super. Is that what you had in mind? The only disadvantage that I see is that it's going to be slightly less efficient.

Reply:
Actually, this way may take even longer to see the result than handling the event, since the event is fired/handled synchronously I think, isn't it?

Reply:
You're right about that @brianh5, yeah. However I am noticing that there aren't that many places where we use these events. The pinned event is used nowhere.. so we could potentially axe it. And the selected event is only used to update the counters for the off-stage participant high-layer suspension functionality.

Do we have a new way of doing events? If so I can migrate the selected event to that new way without too much effort (and also remove the pinned event which is defunct).

Reply:
We don't have any way of doing events, but I do think this method is nice--at least I wouldn't want us to rip it out without taking a deeper look there.